src/components/AmountForm.tsx

```diff
@@ -0,0 +1,241 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {View} from 'react-native';
+import type {NativeSyntheticEvent, TextInput, TextInputSelectionChangeEventData} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Browser from '@libs/Browser';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import getOperatingSystem from '@libs/getOperatingSystem';
+import * as MoneyRequestUtils from '@libs/MoneyRequestUtils';
+import CONST from '@src/CONST';
+import BigNumberPad from './BigNumberPad';
+import FormHelpMessage from './FormHelpMessage';
+import TextInputWithCurrencySymbol from './TextInputWithCurrencySymbol';
+
+type AmountFormProps = {
+    /** Amount supplied by the FormProvider */
+    value?: string;
+
+    /** Currency supplied by user */
+    currency?: string;
+
+    /** Tells how many extra decimal digits are allowed. Default is 0. */
+    extraDecimals?: number;
+
+    /** Error to display at the bottom of the component */
+    errorText?: string;
+
+    /** Callback to update the amount in the FormProvider */
+    onInputChange?: (value: string) => void;
+
+    /** Fired when back button pressed, navigates to currency selection page */
+    onCurrencyButtonPress?: () => void;
+};
+
+/**
+ * Returns the new selection object based on the updated amount's length
+ */
+const getNewSelection = (oldSelection: {start: number; end: number}, prevLength: number, newLength: number) => {
+    const cursorPosition = oldSelection.end + (newLength - prevLength);
+    return {start: cursorPosition, end: cursorPosition};
+};
+
+const AMOUNT_VIEW_ID = 'amountView';
+const NUM_PAD_CONTAINER_VIEW_ID = 'numPadContainerView';
+const NUM_PAD_VIEW_ID = 'numPadView';
+
+function AmountForm(
+    {value: amount, currency = CONST.CURRENCY.USD, extraDecimals = 0, errorText, onInputChange, onCurrencyButtonPress}: AmountFormProps,
+    forwardedRef: ForwardedRef<TextInput>,
+) {
+    const styles = useThemeStyles();
+    const {toLocaleDigit, numberFormat} = useLocalize();
+
+    const textInput = useRef<TextInput | null>(null);
+
+    const decimals = CurrencyUtils.getCurrencyDecimals(currency) + extraDecimals;
+    const currentAmount = useMemo(() => (typeof amount === 'string' ? amount : ''), [amount]);
+
+    const [shouldUpdateSelection, setShouldUpdateSelection] = useState(true);
+
+    const [selection, setSelection] = useState({
+        start: currentAmount.length,
+        end: currentAmount.length,
+    });
+
+    const forwardDeletePressedRef = useRef(false);
+
+    /**
+     * Event occurs when a user presses a mouse button over an DOM element.
+     */
+    const onMouseDown = (event: React.MouseEvent, ids: string[]) => {
```

Comment:
Please rename this method to `focusTextInput`

Reply:
Yeah, you are right, but for our naming conventions, all functions should be named for what they do, not what they handle (the exception is when naming prop callback functions). All the rest of the methods in this component are following the proper conventions, so this one stands out.